packages/theme-a11y/theme-a11y.test.js

```diff
@@ -0,0 +1,132 @@
+
+    focusableElement.tabIndex = 3;
+    pageLinkFocus(focusableElement);
+
+    expect(parseInt(focusableElement.dataset.tabIndex)).toBe(3);
+  });
+
+  test("adds a focus state", () => {
+    const focusableElement = document.getElementById("focusableElement");
+
+    pageLinkFocus(focusableElement);
+
+    expect(document.activeElement).toBe(focusableElement);
+  });
+
+  test("adds the default class name 'js-focus-hidden'", () => {
+    const focusableElement = document.getElementById("focusableElement");
+    const defaultClass = "js-focus-hidden";
+    pageLinkFocus(focusableElement);
+
+    expect(focusableElement.classList.contains(defaultClass)).toBeTruthy();
+  });
+
+  test("adds an overrided class name", () => {
+    const focusableElement = document.getElementById("focusableElement");
+    const customClass = "custom-class";
+    pageLinkFocus(focusableElement, { className: customClass });
+
+    expect(focusableElement.classList.contains(customClass)).toBeTruthy();
+  });
+
+  test("resets the element to its original state on blur", () => {
+    const focusableElement = document.getElementById("focusableElement");
+    const nonFocusableElement = document.getElementById("nonFocusableElement");
+    const clone = focusableElement.cloneNode(true);
+
+    pageLinkFocus(focusableElement);
+    pageLinkFocus(nonFocusableElement);
+
+    expect(focusableElement).toEqual(clone);
+  });
+});
+
+describe("focusHash()", () => {
+  beforeEach(() => {
+    document.body.innerHTML =
+      '<div id="nonFocusableElement"></div>' +
+      '<button id="focusableElement" tabIndex=3 class="myClass">';
+  });
+
+  test("is a function exported by a11y.js", () => {
+    expect(typeof focusHash).toBe("function");
+  });
+
+  test("focuses an element whose ID is specified in the URL hash", () => {
+    const focusableElement = document.getElementById("focusableElement");
+
+    window.location.hash = "focusableElement";
+    focusHash();
+
+    expect(document.activeElement).toBe(focusableElement);
+  });
+
+  test("does not change the focus if element specified does not exist", () => {
+    const focusableElement = document.getElementById("focusableElement");
+
+    window.location.hash = "otherElement";
+    pageLinkFocus(focusableElement);
+    focusHash();
+
+    expect(document.activeElement).toBe(focusableElement);
+  });
+});
+
+describe("bindInPageLinks()", () => {
+  beforeEach(() => {
+    document.body.innerHTML =
+      '<a id="link" href="#title"></a>' + '<h1 id="title">Title</h1>';
+  });
+
+  test("is a function exported by a11y.js", () => {
+    expect(typeof bindInPageLinks).toBe("function");
+  });
+
+  test("adds an event handler that focuses the element referred to in an <a> element w/ a href='#...' when it is clicked", () => {
+    const link = document.getElementById("link");
+    const title = document.getElementById("title");
+
+    bindInPageLinks();
+
+    link.click();
+    expect(document.activeElement).toBe(title);
+  });
```

Comment:
It should there should be a test for the opposite behaviour :

```
beforeEach(() => {
    document.body.innerHTML = 
    `
        <a id="link" href="#title"></a>
        <a id="regularLink" href="/test">Regular link</a>
        <h1 id="title">Title</h1>
    `;
});

test("does not focus the element referred to an <a> element without a href='#...' when it is clicked", () => {
    const regularLink = document.getElementById("regularLink");
    const title = document.getElementById("title");

    bindInPageLinks();

    link.click();
    expect(document.activeElement).toBe(regularLink);
});
```

Reply:
Difficult to test that behaviour in JSDOM. Going to just test green-path for now.